File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
	String[] words=s.split(" ");
    StringBuilder sb=new StringBuilder(words.length);
    for(int i=words.length-1;i>=0;i--){
        if(words[i].length()!=0){
            if(sb.length()!=0){
                sb.append(" ");
            }
            sb.append(words[i]);
        }
    }
    return sb.toString();
    }
}

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string str) {
    vector<string> words;
	string curr;
	for(char c:str){
		if(c==' '){
			if(!curr.empty()){
				words.push_back(curr);
                curr.clear();
			}
		}
		else{
			curr+=c;
		}
	}
	if(!curr.empty()){       //Handle last words
		words.push_back(curr);
	}
	int low=0;
	int high=words.size()-1;
	while(low<=high){
		swap(words[low],words[high]);
		low++;
		high--;
	}
	string rev;
	for(string word:words){
		rev+=word+' ';
	}
	
	if(!rev.empty()){
		rev.pop_back();
	}
	return rev;
    }
};